src/identity-providers/routes/IdentityProviderCreate.ts

```diff
@@ -0,0 +1,22 @@
+import type { LocationDescriptorObject } from "history";
+import { generatePath } from "react-router-dom";
+import type { RouteDef } from "../../route-config";
+import { AddIdentityProvider } from "../add/AddIdentityProvider";
+
+export type IdentityProviderCreateParams = {
+  realm: string;
+  providerId?: string;
```

Comment:
The `providerId` field here should be made required like in the path of the route.
```suggestion
  providerId: string;
```